Fix issue 8376 - Scrolling in a short email may trigger left/right swipe

joohnq opened the pull request:
Fixes #8376 

I used the fling listener to reduce the sensitivity on the X axis. This way, it's less likely that the user will accidentally swipe to another email while scrolling.

Now, I was thinking of creating an option that the user can enable or disable. What do you think?

joohnq commented:
I’d like to know if there’s a global object or class in the codebase that stores constants or configuration values. For example, I added a value that multiplies the X velocity, and I want to know if this kind of value should stay hardcoded or be placed in a shared/configurable location.

wmontwe commented on legacy/ui/legacy/src/main/java/com/fsck/k9/ui/messageview/MessageViewContainerFragment.kt:
```diff
@@ -109,6 +115,19 @@ class MessageViewContainerFragment : Fragment() {
             },
         )
 
+        val recyclerView = viewPager.getChildAt(0) as? RecyclerView
+        recyclerView?.let { rv ->
+            // Reduce horizontal fling speed to make page swipes less sensitive and prevent accidental swipes.
+            rv.onFlingListener = object : RecyclerView.OnFlingListener() {
+                override fun onFling(velocityX: Int, velocityY: Int): Boolean {
+                    val scaledVelocity = (velocityX * VIEW_PAGER_FLING_VELOCITY_SCALE).toInt()
+                    val adjustedVelocityX = scaledVelocity.coerceIn(VIEW_PAGER_MIN_FLING_VELOCITY_X, VIEW_PAGER_MAX_FLING_VELOCITY_X)
+
+                    return rv.fling(adjustedVelocityX, velocityY)
+                }
+            }
+        }
+
```
Unfortunately, `ViewPager2` does not provide an official API for customizing swipe behavior. As a result, many workarounds rely on reflection to modify internal parameters controlling swipe gestures, which may not be the most reliable or maintainable approach.

A more viable alternative is creating a custom implementation of `CustomViewPager2` by overriding the `onTouchEvent` method, allowing for adjustments to the coordinate transformations to reduce sensitivity.

But as `ViewPager2` wraps a `RecyclerView`, using the built-in features of the `ItemTouchHelper` can simplify managing swipe behavior. This helper provides robust event handling that can be used to react to specific touch events only (e.g., swipe vs. move, direction left or right).

However, I've also explored using the `OnFlingListener` approach with varying values, but unfortunately, these efforts have shown either unsatisfactory results or caused application crashes.

joohnq commented:
All right, based on your suggestion, I’ve updated the code and split this feature into a separate function. If the function structure doesn’t look good to you, feel free to let me know.

wmontwe commented:
Thanks for the update, as mentioned in the issue comments already, we would not like to introduce another option to the settings.

wmontwe reviewed (approved):
Thank you!
